### PR TITLE
Application Delivery Break Out

### PR DESCRIPTION
**Pull request by brsteph**
This PR contains the following changes:

 Modifies the existing ALZ checklist to reduce the number of Application Delivery items:

- Removes the most of the existing D01 and D10 series of recommendations from the Azure Landing Zone Checklist.
- Adds a new set of recommendations in D01 that are more generic and focused on having a spoke-based plan for application delivery, keeping the generic recommendations for doing app delivery within landing zones and having a DDoS plan in place.  These recommendations point to the Application Delivery checklist below.
- Replaces the F01 series of recommendations for WAF logs to be more generic (not front door specific)

Creates a new Application Delivery checklist that contains the items removed from the ALZ checklist.

### CODE REVIEW

File: checklists/alz_checklist.en.json [hunk 1]
```diff
@@ -564,12 +564,12 @@
         {
             "category": "Network Topology and Connectivity",
             "subcategory": "App delivery",
-            "text": "If you use customer-managed TLS certificates with Azure Front Door, use the 'Latest' certificate version. Reduce the risk of outages caused by manual certificate renewal.",
+            "text": "Develop a plan for delivering application content from your Workload spokes, using the Application Delivery checklist as a guide.",
```
Comment (igorjnzl) [inline checklists/alz_checklist.en.json:567]: Are we able to change the wording from "Application Delivery checklist" to recommendations from the link? IE. Leverage App GW / AFG for secure delivery of web applications. Maybe we can split into multiple checklist items

Comment (igorjnzl) [reply checklists/alz_checklist.en.json:567]: More detailed and in-depth application delivery checks can go in separate list; however, we still want to capture some high-level recommendations as part of the ALZ checklist to align to our CAF guidance.  We can add a few items to capture the recommendations from:

https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/plan-for-app-delivery

Comment (brsteph) [reply checklists/alz_checklist.en.json]: Updated the text, let me know if that work!

**Review by igorjnzl — APPROVED** [review]
Thanks for the updates

**Review by erjosito — APPROVED** [review]

**Pull request merged**